src/systems/movement.rs

```diff
@@ -0,0 +1,74 @@
+use amethyst::core::cgmath::{Angle, InnerSpace, Point2, Quaternion, Rad, Rotation3, Vector3};
+use amethyst::core::specs::prelude::{Component, Entity, Entities,Join, Read, System, WriteStorage};
+use amethyst::core::timing::Time;
+use amethyst::core::transform::Transform;
+use amethyst::ecs::storage::DenseVecStorage;
+
+pub struct MovementSystem;
+
+impl<'a> System<'a> for MovementSystem {
+    type SystemData = (
+        Entities<'a>,
+        Read<'a, Time>,
+        WriteStorage<'a, Path>,
+        WriteStorage<'a, Transform>,
+    );
+
+    fn run(&mut self, (entities, time, mut movement_intents, mut transforms): Self::SystemData) {
+        let mut removable_paths:Vec<Entity> = vec![];
+
+        for (entity, mut path, mut transform) in (&entities, &mut movement_intents, &mut transforms).join() {
+
+
+            let next_destination = path[0];
+            let next_destination = Vector3::from((next_destination.x, 0., next_destination.y));
+
+            // shouldn't this be reversed?
+            // BUT IT WORKS, SO IT STAYS until there is a solution :)
+            let direction = transform.translation- next_destination ;
+
+            (*transform).rotation = Quaternion::from_angle_y(Rad::atan2( direction.x,direction.z));
+
+            transform.move_forward(time.delta_seconds());
+
+            if (next_destination - transform.translation).magnitude() < time.delta_seconds() {
+                path.remove(0);
+            }
+
+            if path.is_empty() {
+                removable_paths.push(entity);
+            }
```

Comment:
have you tried to remove the element while iterating over the storage?

Reply:
we are iterating over the element we want to delete from? thats against every borrowing rule rust has